tasks/docker.rb

```diff
@@ -80,19 +80,29 @@ def fix_ssh(distro, version, container)
 
 def get_image_os_release_facts(image)
   os_release_facts = {}
-  os_release = run_local_command("docker run --rm #{image} cat /etc/os-release")
-  # The or-release file is a newline-separated list of environment-like
-  # shell-compatible variable assignments.
-  re = '^(.+)=(.+)'
-  os_release.each_line do |line|
-    line = line.strip || line
-    next unless !line.nil? && !line.empty?
+  begin
+    os_release = run_local_command("docker run --rm #{image} cat /etc/os-release")
+    # The or-release file is a newline-separated list of environment-like
+    # shell-compatible variable assignments.
+    re = '^(.+)=(.+)'
+    os_release.each_line do |line|
+      line = line.strip || line
+      next unless !line.nil? && !line.empty?
 
-    _, key, value = line.match(re).to_a
-    # The values seems to be quoted most of the time, however debian only quotes
-    # some of the values :/.  Parse it, as if it was a JSON string.
-    value = JSON.parse(value) unless value[0] != '"'
-    os_release_facts[key] = value
+      _, key, value = line.match(re).to_a
+      # The values seems to be quoted most of the time, however debian only quotes
+      # some of the values :/.  Parse it, as if it was a JSON string.
+      value = JSON.parse(value) unless value[0] != '"'
+      os_release_facts[key] = value
+    end
+  rescue => e
+    # fall through to parsing the id and version from the image if it doesn't have `/etc/os-release`
+    puts({ _error: { kind: 'provision/docker_failure', msg: e.message, backtrace: e.backtrace } }.to_json)
```

Comment:
have you tested this? Won't outputting the error here will interfere with the regular results from the task?

Reply:
Have double checked the changes by adding them into one of my local forks.
Removing the above line as having any outs here prevents the machine name from being output as a result, i.e.:
`, litmusimage/centos:6`